### PR TITLE
Changes for issue 5569

### PR DESCRIPTION
**Pull request by kevinfason**
Per issue 5569 I made some proposed changes to the two files around NTP network servers. I broke out the changes in two phases. One is to simply parse the GUI entered servers to determine if they are a public pool server or not and set them correctly in the ntpd.conf file. Secondly to add setting in GUI to set maxcount. This tells ntpd how many concurrent servers to maintain. Due to the pool directive, status page will not show only the servers declared on the network tab (GPS,PPS unimpacted).  With the use of pool servers I felt it would be useful for users to set this if they wish.

### CODE REVIEW

**Comment by fichtner** [comment]
@kevinfason sorry I lost track of this.. is this tested/finished?

File: src/etc/inc/plugins.inc.d/ntpd.inc [hunk 1]
```diff
@@ -200,7 +200,17 @@ function ntpd_configure_do($verbose = false)
         $ntpcfg .= '12';
     }
     $ntpcfg .= "\n";
-
+
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/ntpd.inc:203]:
```suggestion

```

File: src/etc/inc/plugins.inc.d/ntpd.inc [hunk 2]
```diff
@@ -200,7 +200,16 @@ function ntpd_configure_do($verbose = false)
         $ntpcfg .= '12';
     }
     $ntpcfg .= "\n";
-
+    /* Add Max Clock Count */
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/ntpd.inc:203]:
```suggestion

    /* Add Max Clock Count */
```

File: src/etc/inc/plugins.inc.d/ntpd.inc [hunk 3]
```diff
@@ -323,7 +333,12 @@ function ntpd_configure_do($verbose = false)
     $ntpcfg .= "\n\n# Upstream Servers\n";
     /* foreach through ntp servers and write out to ntpd.conf */
     foreach (explode(' ', $config['system']['timeservers']) as $ts) {
+        /* Determine if Network Time Server is from the NTP Pool or not */
+        if (preg_match("/\.pool\.ntp\.org$/", $ts)) {
+        $ntpcfg .= "pool {$ts}";
+        } else {
         $ntpcfg .= "server {$ts}";
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/ntpd.inc:340]:
```suggestion
            $ntpcfg .= "pool {$ts}";
        } else {
            $ntpcfg .= "server {$ts}";
```

File: src/etc/inc/plugins.inc.d/ntpd.inc [hunk 4]
```diff
@@ -412,6 +427,26 @@ function ntpd_configure_do($verbose = false)
     }
     if (empty($config['ntpd']['notrap'])) { /*note: this one works backwards */
         $ntpcfg .= ' notrap';
+    }
+            /* Access restrictions for Pool */
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/ntpd.inc:431]: is this necessary? for mixed scenarios?

File: src/www/services_ntpd.php [hunk 5]
```diff
@@ -91,6 +92,9 @@
     if (!empty($pconfig['orphan']) && ($pconfig['orphan'] < 0 || $pconfig['orphan'] > 15 || !is_numeric($pconfig['orphan']))) {
         $input_errors[] = gettext("Orphan mode must be a value between 0..15");
     }
+    if (!empty($pconfig['maxclock']) && ($pconfig['maxclock'] < 0 || $pconfig['maxclock'] > 99 || !is_numeric($pconfig['maxclock']))) {
```
Comment (fichtner) [inline src/www/services_ntpd.php:95]: "0" will not pass !empty() check

**Comment by kevinfason** [comment]
> @kevinfason sorry I lost track of this.. is this tested/finished?

It is ready to go IMO with some potential cleanup by someone better skilled than I. I have been using it on 4 systems since I made the proposed changes.

**Comment by fichtner** [comment]
I've taken your code and improved it a bit: 7366d785a62d

**Pull request closed without merging**